sdk/maps/Azure.ResourceManager.Maps/tests/MapsManagementTestBase.cs

```diff
@@ -25,17 +32,65 @@ protected MapsManagementTestBase(bool isAsync)
         }
 
         [SetUp]
-        public void CreateCommonClient()
+        public async Task CreateCommonClient()
         {
             Client = GetArmClient();
+            DefaultSubscription = await Client.GetDefaultSubscriptionAsync();
         }
 
-        protected async Task<ResourceGroupResource> CreateResourceGroup(SubscriptionResource subscription, string rgNamePrefix, AzureLocation location)
+        protected async Task<ResourceGroupResource> CreateResourceGroupAsync()
         {
-            string rgName = Recording.GenerateAssetName(rgNamePrefix);
-            ResourceGroupData input = new ResourceGroupData(location);
-            var lro = await subscription.GetResourceGroups().CreateOrUpdateAsync(WaitUntil.Completed, rgName, input);
+            string rgName = Recording.GenerateAssetName(ResourceGroupNamePrefix);
+            ResourceGroupData input = new ResourceGroupData(DefaultLocation);
+            var lro = await DefaultSubscription.GetResourceGroups().CreateOrUpdateAsync(WaitUntil.Completed, rgName, input);
             return lro.Value;
         }
+
+        protected MapsAccountData GetDefaultMapsAccountData()
+        {
+            var account = new MapsAccountData(DefaultLocation, new MapsSku(Name.S0))
+            {
+                Tags = { { "key1", "value1" }, { "key2", "value2" } }
+            };
+
+            return account;
+        }
+
+        public void VerifyAccountProperties(MapsAccountData account, bool useDefaults, Name skuName)
+        {
+            Assert.NotNull(account.Id);
+            Assert.NotNull(account.Location);
+            Assert.NotNull(account.Name);
+
+            Assert.NotNull(account.Sku);
+            Assert.NotNull(account.Sku.Tier);
+
+            if (useDefaults)
+            {
+                Assert.AreEqual(DefaultLocation, account.Location);
+                Assert.AreEqual(Name.S0, account.Sku.Name);
+
+                Assert.NotNull(account.Tags);
+                Assert.NotNull(account.Properties.UniqueId);
+                Assert.AreEqual(2, account.Tags.Count);
+                Assert.AreEqual("value1", account.Tags["key1"]);
+                Assert.AreEqual("value2", account.Tags["key2"]);
+            }
+            else
+            {
+                Assert.AreEqual(skuName, account.Sku.Name);
+                Assert.AreEqual(DefaultLocation, account.Location);
+            }
+        }
+
+        public async Task<string> CreateDefaultMapsAccount(MapsAccountCollection mapCollection, string rgname)
```

Comment:
parameter `rgname` is not used. We could remove it

Reply:
BTW it should be better if we could return the MapsAccountResource instead of only returning the name

Reply:
Fixed